src/SprykerSdk/Integrator/Executor/ManifestExecutor.php

```diff
@@ -0,0 +1,163 @@
+    {
+        $this->assertModuleData($moduleTransfers);
+
+        $manifests = $this->manifestReader->readManifests($moduleTransfers);
+
+        $sprykerLock = $this->sprykerLockReader->getLockFileData();
+
+        $unappliedManifests = $this->findUnappliedManifests($manifests, $sprykerLock);
+
+        if (!$unappliedManifests) {
+            return 0;
+        }
+
+        if (!$inputOutput->confirm('There are unapplied manifests found for your modules. Do you want to apply them?')) {
+            return 0;
+        }
+
+        $GLOBALS['IO'] = $inputOutput;
+
+        foreach ($unappliedManifests as $moduleName => $moduleManifests) {
+            foreach ($moduleManifests as $manifestType => $unappliedManifestByType) {
+                $manifestExecutor = $this->resolveExecutor($manifestType);
+
+                foreach ($unappliedManifestByType as $manifestHash => $unappliedManifest) {
+                    if ($manifestExecutor->apply($unappliedManifest, $moduleName, $inputOutput, $isDry)) {
+                        $sprykerLock[$moduleName][$manifestType][$manifestHash] = $unappliedManifest;
+                    }
+                }
+            }
+        }
+        if ($isDry) {
+            return 0;
+        }
+
+        return $this->sprykerLockWriter->storeLock($sprykerLock);
+    }
+
+    /**
+     * @param string[][][] $manifests
+     * @param string[][] $sprykerLock
+     *
+     * @return string[][][][]
+     */
+    protected function findUnappliedManifests(array $manifests, array $sprykerLock): array
+    {
+        $unappliedManifests = [];
+        foreach ($manifests as $moduleName => $manifestList) {
+            foreach ($manifestList as $manifestType => $moduleManifests) {
+                foreach ($moduleManifests as $moduleManifest) {
+                    $manifestHash = sha1(json_encode($moduleManifest) . $manifestType . $moduleName);
+                    if (isset($sprykerLock[$moduleName][$manifestType][$manifestHash])) {
+                        continue;
+                    }
+
+                    $unappliedManifests[$moduleName][$manifestType][$manifestHash] = $moduleManifest;
+                }
+            }
+        }
+
+        return $unappliedManifests;
+    }
+
+    /**
+     * @param string $manifestType
+     *
+     * @throws \RuntimeException
+     *
+     * @return \SprykerSdk\Integrator\ManifestStrategy\ManifestStrategyInterface
+     */
+    protected function resolveExecutor(string $manifestType): ManifestStrategyInterface
+    {
+        foreach ($this->manifestExecutors as $manifestExecutor) {
+            if ($manifestType === $manifestExecutor->getType()) {
+                return $manifestExecutor;
+            }
+        }
+
+        throw new RuntimeException("Executor $manifestType not found");
+    }
+
+    /**
+     * @param \SprykerSdk\Integrator\Transfer\ModuleTransfer[] $moduleTransfers
```

Comment:
would be nice to use new syntax for docblocks like array<\SprykerSdk\Integrator\Transfer\ModuleTransfer>